### PR TITLE
Replace a no-name convnet with LeNet-5

### PR DESCRIPTION
**Pull request by ozabluda**
A good place put LeNet-5 into the codebase (currently absent), as well as the reference to the paper which introduced MNIST. Achieves the same accuracy in the same number of epochs as the no-name one it is replacing (and the same one as in the paper). Has fewer parameters, faster, less memory usage. Simpler (no dropout layers), which it doesn't benefit from, because LeNet-5 has 61,706 parameters, almost the same as the number of training samples.

### CODE REVIEW

**Comment by ygorcanalli** [comment]
The "no-name" network is simpler than LeNet5, easier to understand how to use Keras. Would be a good thing another example with LeNet5, maybe in contrib.

**Comment by fchollet** [comment]
LeNet does not follow modern best practices, and the current net works just fine. The goal here is just to provide a demo of how to use Keras, not to illustrate specific papers.

Sorry for not reviewing this earlier.
